domain/schema/model/sql/0015-charm.sql

```diff
@@ -307,17 +307,17 @@ CREATE INDEX idx_charm_relation_charm
 ON charm_relation (charm_uuid);
 
 CREATE TABLE charm_extra_binding (
+    uuid TEXT NOT NULL PRIMARY KEY,
```

Comment:
I'm not sure about synthetic uuids, but I can see foreign keys become easier.

You would need to update the charm code to ensure that we set it correctly.

Reply:
Agreed.